diff --git a/diffkemp/diffkemp.py b/diffkemp/diffkemp.py
index 2dca570..3dbf40c 100644
--- a/diffkemp/diffkemp.py
+++ b/diffkemp/diffkemp.py
@@ -379,11 +379,14 @@ def compare(args):
 
     config = Config.from_args(args)
     result = Result(Result.Kind.NONE, args.snapshot_dir_old,
-                    args.snapshot_dir_old, start_time=default_timer())
+                    args.snapshot_dir_old, start_time=default_timer(),
+                    most_outer=True)
 
     for group_name, group in sorted(config.snapshot_first.fun_groups.items()):
         group_printed = False
 
+        group_result = Result(Result.Kind.NONE, group_name, group_name)
+
         # Set the group directory
         if output_dir is not None and group_name is not None:
             group_dir = os.path.join(output_dir, group_name)
@@ -410,7 +413,7 @@ def compare(args):
 
             # Check if the module exists in both snapshots
             if old_fun_desc.mod is None or new_fun_desc.mod is None:
-                result.add_inner(Result(Result.Kind.UNKNOWN, fun, fun))
+                group_result.add_inner(Result(Result.Kind.UNKNOWN, fun, fun))
                 if group_name is not None and not group_printed:
                     print("{}:".format(group_name))
                     group_printed = True
@@ -440,7 +443,7 @@ def compare(args):
                     else:
                         fun_result.kind = Result.Kind.EQUAL
 
-                result.add_inner(fun_result)
+                group_result.add_inner(fun_result)
 
                 # Printing information about failures and non-equal functions.
                 if fun_result.kind in [Result.Kind.NOT_EQUAL,
@@ -482,9 +485,11 @@ def compare(args):
             old_fun_desc.mod.clean_module()
             new_fun_desc.mod.clean_module()
             LlvmModule.clean_all()
+        result.add_inner(group_result)
     # Create yaml output
     if output_dir is not None and os.path.isdir(output_dir):
         old_dir_abs = os.path.join(os.path.abspath(args.snapshot_dir_old), "")
         new_dir_abs = os.path.join(os.path.abspath(args.snapshot_dir_new), "")
         yaml_output = YamlOutput(snapshot_dir_old=old_dir_abs,
+                                 kind=config.snapshot_first.list_kind,
                                  snapshot_dir_new=new_dir_abs, result=result)
diff --git a/diffkemp/output.py b/diffkemp/output.py
index 8af8a53..7b9ceb6 100644
--- a/diffkemp/output.py
+++ b/diffkemp/output.py
@@ -8,10 +8,11 @@ import yaml
 class YamlOutput:
     """Yaml representation of compare result."""
 
-    def __init__(self, snapshot_dir_old, snapshot_dir_new, result):
+    def __init__(self, snapshot_dir_old, snapshot_dir_new, result, kind):
         self.old_dir = snapshot_dir_old
         self.new_dir = snapshot_dir_new
         self.result = result
+        self.kind = kind
         self.output = {}
         # Sets with symbol names
         self.function_names = set()
@@ -29,15 +30,22 @@ class YamlOutput:
         """Creates yaml representation of compare result."""
         self.output["old-snapshot"] = self.old_dir
         self.output["new-snapshot"] = self.new_dir
+        self.output["results"] = []
+        self.output["definitions"] = {}
+        self.output["comparison-kind"] = self.kind
 
         self._create_results()
-        self._create_definitions()
 
     def _create_results(self):
         """Creates call stacks of not-equal functions."""
         # list of all not-equal functions callstacks
+        for group_name, group_results in self.result.inner.items():
+            self._create_group_results(group_results, group_name)
+            self._create_group_definitions(group_results.graph)
+
+    def _create_group_results(self, group_results, group_name):
         results = []
-        for fun_name, fun_result in self.result.inner.items():
+        for fun_name, fun_result in group_results.inner.items():
             if fun_result.kind != Result.Kind.NOT_EQUAL:
                 continue
             self.function_names.add(fun_name)
@@ -71,19 +79,20 @@ class YamlOutput:
                 "function": fun_name,
                 "diffs": diffs
             })
-        self.output["results"] = results
+        self.output["results"].extend(results)
 
-    def _create_definitions(self):
-        self.output["definitions"] = {}
-        self._create_function_defs()
-        self._create_type_defs()
+    def _create_group_definitions(self, graph):
+        self._create_function_defs(graph)
+        self._create_type_defs(graph)
 
-    def _create_function_defs(self):
+    def _create_function_defs(self, graph):
+        if graph is None:
+            return
         definitions = {}
         for name in self.function_names:
-            if name not in self.result.graph.vertices:
+            if name not in graph.vertices:
                 continue
-            vertex = self.result.graph[name]
+            vertex = graph[name]
             definition = {
                 "kind": "function",
                 "old": self._create_def_info(vertex.lines[0],
@@ -101,17 +110,18 @@ class YamlOutput:
             definitions[name] = definition
         self.output["definitions"].update(definitions)
 
-    def _create_type_defs(self):
-        definitions = {}
+    def _create_type_defs(self, graph):
         """Returns definitions of types.
         :param type_names: Set of tuples (type name, parent name),
                            parent name - name of function in which type is used
         """
+        if graph is None:
+            return
         definitions = {}
         for type_name, parent_name in self.type_names:
-            if parent_name not in self.result.graph.vertices:
+            if parent_name not in graph.vertices:
                 continue
-            vertex = self.result.graph[parent_name]
+            vertex = graph[parent_name]
             for non_fun in vertex.nonfun_diffs:
                 if non_fun.name == type_name and \
                         isinstance(non_fun, ComparisonGraph.TypeDiff):
diff --git a/diffkemp/semdiff/result.py b/diffkemp/semdiff/result.py
index 7ac2423..92b608f 100644
--- a/diffkemp/semdiff/result.py
+++ b/diffkemp/semdiff/result.py
@@ -140,7 +140,7 @@ class Result:
             self.covered = covered
 
     def __init__(self, kind, first_name, second_name, start_time=None,
-                 stop_time=None):
+                 stop_time=None, most_outer=False):
         self.kind = kind
         self.first = Result.Entity(first_name)
         self.second = Result.Entity(second_name)
@@ -150,6 +150,7 @@ class Result:
         self.inner = dict()
         self.start_time = start_time
         self.stop_time = stop_time
+        self.most_outer = most_outer
 
     def __str__(self):
         return str(self.kind)
@@ -290,6 +291,18 @@ class Result:
 
     def report_stat(self, show_errors=False, extended_stat=False):
         """Reports all statistics."""
+        # Backward compatibility
+        if self.most_outer:
+            r = Result(self.kind, self.first.name, self.second.name,
+                       start_time=self.start_time,
+                       stop_time=self.stop_time)
+            # over groups
+            for inner in self.inner.values():
+                for fun in inner.inner.values():
+                    r.add_inner(fun)
+            r.report_stat(show_errors, extended_stat)
+            return
+
         self.report_symbol_stat(show_errors)
         if extended_stat:
             print("")
